core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -1763,6 +1775,32 @@ object Chunk extends ChunkFactory with ChunkPlatformSpecific {
     override private[zio] def reverseArrayIterator[A1 >: Boolean]: Iterator[Array[A1]] =
       arrayIterator
 
+    override def &(that: Chunk[Boolean])(implicit ev: Boolean <:< Boolean): Chunk[Boolean] = that match {
+      case thatBitChunk: BitChunk => byteWiseOp(thatBitChunk)(_ => 0, _ => 0)((a: Byte, b: Byte) => (a & b).toByte)
+      case that                   => super.&(that)
+    }
+
+    override def |(that: Chunk[Boolean])(implicit ev: Boolean <:< Boolean): Chunk[Boolean] = that match {
+      case thatBitChunk: BitChunk => byteWiseOp(thatBitChunk)(identity, identity)((a: Byte, b: Byte) => (a | b).toByte)
+      case that                   => super.|(that)
+    }
+
+    override def ^(that: Chunk[Boolean])(implicit ev: Boolean <:< Boolean): Chunk[Boolean] = that match {
+      case thatBitChunk: BitChunk => byteWiseOp(thatBitChunk)(identity, identity)((a: Byte, b: Byte) => (a ^ b).toByte)
+      case that                   => super.^(that)
+    }
+
+    override def unary_~(implicit ev: Boolean <:< Boolean): BitChunk =
+      BitChunk(bytes.mapChunk(b => (~b).toByte), minBitIndex, maxBitIndex)
+
+    private def byteWiseOp(
```

Comment:
I don't think this is going to correctly handle cases where one of the chunk has been sliced so the bit indices don't line up. Can we add some tests for that?

Reply:
You're right, I've added tests.